### PR TITLE
[DOCS-1778] Manual serverless trace propagation

### PR DESCRIPTION
**Pull request by Hesperide**
Documenting how to manually propagate serverless trace context through non-HTTP requests.

Staging link: https://docs-staging.datadoghq.com/alex.cuoci/node-py-manual-trace-propagation/serverless/distributed_tracing

### CODE REVIEW

File: content/en/serverless/distributed_tracing/serverless_trace_propagation.md [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+  text: "Live Search"
+---
+
+
+{{< img src="serverless/lambda-non-http-trace.png" alt="Serverless Distributed Non-HTTP Trace"  style="width:100%;" >}}
+
+## Required Setup
+
+Additional instrumentation is sometimes required to see a single, connected trace in Node and Python serverless applications asynchronously triggering Lambda functions. If you are just getting started with monitoring serverless applications in Datadog, first [follow our main installation steps][1] and [read this page on choosing your tracing library][2]. Once you are sending traces from your Lambda functions to Datadog using the [Datadog Lambda Library][3], you may want to follow these steps to connect traces between two Lambda functions in cases such as:
+- Asynchronously triggering Lambda functions via SNS, Kinesis or EventBridge
```
Comment (tianchu) [inline content/en/serverless/distributed_tracing/serverless_trace_propagation.md:19]: I think it's worth mentioning what's supported out of box, so people don't confuse themselves and spend unnecessary effort on implementing this.

File: content/en/serverless/distributed_tracing/serverless_trace_propagation.md [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+In Python, you can use the `get_dd_trace_context` helper function to easily pass tracing context to outgoing events:
+
+```py
+import json
+import boto3
+import os
+
+from datadog_lambda.tracing import get_dd_trace_context  # Datadog tracing helper function
+
+SNS_CLIENT = boto3.client('sns')
```
Comment (tianchu) [inline content/en/serverless/distributed_tracing/serverless_trace_propagation.md:42]: nits, `sns_client` seems more pythonic here.

File: content/en/serverless/distributed_tracing/serverless_trace_propagation.md [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+
+def handler(event, context):
+    SNS_CLIENT.publish(
+        TopicArn = os.getenv('SNS_ARN'),
```
Comment (tianchu) [inline content/en/serverless/distributed_tracing/serverless_trace_propagation.md:46]: perhaps `os.getenv('SNS_ARN')` should just be a placeholder, so people don't get confused and thought they must set this env var?

File: content/en/serverless/distributed_tracing/serverless_trace_propagation.md [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+        Message = 'Asynchronously invoking a Lambda function with SNS.',
+        MessageAttributes = {
+            '_datadog': {
+                'DataType': 'String',
+                'StringValue': json.dumps(get_dd_trace_context()) # Includes trace context in outgoing message to SNS topic.
+            },
+        },
+    )
+```
```
Comment (tianchu) [inline content/en/serverless/distributed_tracing/serverless_trace_propagation.md:55]: Any reason why don't we give out examples for EventBridge and Kinesis? I felt that in reality most of developers would give us or feel frustrated if there isn't an example to copy&paste. Also feels weird when we have sample extractor code for those.

File: content/en/serverless/distributed_tracing/serverless_trace_propagation.md [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+{{< img src="serverless/lambda-non-http-trace.png" alt="Serverless Distributed Non-HTTP Trace"  style="width:100%;" >}}
+
+## Required Setup
+
+Additional instrumentation is sometimes required to see a single, connected trace in Node and Python serverless applications asynchronously triggering Lambda functions. If you are just getting started with monitoring serverless applications in Datadog, first [follow our main installation steps][1] and [read this page on choosing your tracing library][2]. Once you are sending traces from your Lambda functions to Datadog using the [Datadog Lambda Library][3], you may want to follow these steps to connect traces between two Lambda functions in cases such as:
+- Asynchronously triggering Lambda functions via SNS, Kinesis or EventBridge
+- Invoking Lambda functions via non-HTTP protocols such as MQTT
+
+To successfully connect trace context between resources sending traces, you need to:
+- Include Datadog trace context in outgoing events. The outgoing event can originate from a host or Lambda function with `dd-trace` installed.
```
Comment (tianchu) [inline content/en/serverless/distributed_tracing/serverless_trace_propagation.md:23]: The sample injection code we put out only work in lambda, specifically the `get_dd_trace_context` helper function. The original code you had in the sample projects do work for both though, but they are a little bit more complicated.

File: content/en/serverless/distributed_tracing/serverless_trace_propagation.md [hunk 6]
```diff
@@ -0,0 +1,216 @@
...
+  await sns.publish(snsParams).promise();
+  await eventBridge.putEvents(eventBridgeParams).promise()
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+## Extracting Trace Context from Consumer Lambda Functions
+
+To extract the above trace context from the consumer Lambda function, you need to define an extractor function that runs captures trace context before the execution of your Lambda function handler. To do this, configure the `DD_TRACE_EXTRACTOR` environment variable to point to the location of your extractor function. We recommend you place your extractor methods in the same file as your Lambda function handler, or all in one file, as extractors can be re-used across multiple Lambda functions. These extractors are completely customizable to fit any use case, and we included some sample extractors for common use cases in the next section.
```
Comment (tianchu) [inline content/en/serverless/distributed_tracing/serverless_trace_propagation.md:101]: I recommend people put all extractor functions in its own file, which is a good practice (single responsibility principle). Also defining them in the lambda function file can potentially cause circular import problem in some extreme cases.

File: content/en/serverless/distributed_tracing/serverless_trace_propagation.md [hunk 7]
```diff
@@ -0,0 +1,262 @@
...
+  tag: "Documentation"
+  text: "Live Search"
+---
+
+
+{{< img src="serverless/lambda-non-http-trace.png" alt="Serverless Distributed Non-HTTP Trace"  style="width:100%;" >}}
+
+## Required Setup
+
+Additional instrumentation is sometimes required to see a single, connected trace in Node and Python serverless applications asynchronously triggering Lambda functions. If you are just getting started with monitoring serverless applications in Datadog, first [follow our main installation steps][1] and [read this page on choosing your tracing library][2]. Once you are sending traces from your Lambda functions to Datadog using the [Datadog Lambda Library][3], you may want to follow these steps to connect traces between two Lambda functions in cases such as:
```
Comment (apigirl) [inline content/en/serverless/distributed_tracing/serverless_trace_propagation.md:18]:
```suggestion
Additional instrumentation is sometimes required to see a single, connected trace in Node and Python serverless applications asynchronously triggering Lambda functions. If you are just getting started with monitoring serverless applications in Datadog, [follow our main installation steps][1] and [read this page on choosing your tracing library][2]. Once you are sending traces from your Lambda functions to Datadog using the [Datadog Lambda Library][3], you may want to follow these steps to connect traces between two Lambda functions in cases such as:
```

File: content/en/serverless/distributed_tracing/serverless_trace_propagation.md [hunk 8]
```diff
@@ -0,0 +1,262 @@
...
+  const span = tracer.scope().active();
+  const _datadog = {}
+  tracer.inject(span, 'text_map', _datadog)
+
+  // ...
+```
+
+## Extracting Trace Context from Consumer Lambda Functions
+
+To extract the above trace context from the consumer Lambda function, you need to define an extractor function that runs captures trace context before the execution of your Lambda function handler. To do this, configure the `DD_TRACE_EXTRACTOR` environment variable to point to the location of your extractor function (format is `<FILE NAME>.<FUNCTION NAME>`, e.g. `extractors.sns` if the `sns` extract method is in the `extractors.js` file). We recommend you place your extractor methods all in one file, as extractors can be re-used across multiple Lambda functions. These extractors are completely customizable to fit any use case, and we included some sample extractors for common use cases in the next section.
```
Comment (apigirl) [inline content/en/serverless/distributed_tracing/serverless_trace_propagation.md:145]:
```suggestion
To extract the above trace context from the consumer Lambda function, you need to define an extractor function that runs captures trace context before the execution of your Lambda function handler. To do this, configure the `DD_TRACE_EXTRACTOR` environment variable to point to the location of your extractor function (format is `<FILE NAME>.<FUNCTION NAME>`, for example, `extractors.sns` if the `sns` extract method is in the `extractors.js` file). We recommend you place your extractor methods all in one file, as extractors can be re-used across multiple Lambda functions. These extractors are completely customizable to fit any use case, and we included some sample extractors for common use cases in the next section.
```

**Review by apigirl — COMMENTED** [review]
this looks great!

**Review by tianchu — APPROVED** [review]

**Review by apigirl — APPROVED** [review]

**Pull request merged**